readthedocs/core/views/hooks.py

```diff
@@ -75,9 +92,17 @@ def build_branches(project, branch_list):
         not_building - a list of branches that we won't build
     """
     for branch in branch_list:
-        versions = project.versions_from_branch_name(branch)
+        commit = branch['last_commit']
+        versions = project.versions_from_branch_name(branch['name'])
         to_build = set()
         not_building = set()
+        if _contains_skip_mark(commit):
+            log.info('(Branch Build) Skip build %s', project.slug)
+            not_building = {
+                version.slug
+                for version in versions
+            }
+            return (to_build, not_building)
```

Comment:
Here is the logic for skipping the build, not sure if this is the correct place

Reply:
So, if the last commit contains the mark we don't build _any_ version?

I'm confused here since the next few lines there is a logic that can build some version but not some others comming in the same webhook, right? Shouldn't we apply a similar logic here also?

That `to_build` and `not_building` rely on the function `_build_version`. So, this logic shouldn't be in that method? (I don't know, but is has the logging integrated and returns what it needs. On the other hand, it seems that we shouldn't call that method if we already know that we need to skip the version but there are some logic to skip versions inside that function too :/

Reply:
> I'm confused here since the next few lines there is a logic that can build some version but not some others comming in the same webhook, right? Shouldn't we apply a similar logic here also?

Of what I understand, a branch/tag/commit can be tied to various versions so there is only one commit message for all versions, this means that all versions from this branch aren't going to be built. And also is about what you mention, we already know if the version isn't going to be built.